handout/forms/form-builder/form-builder_custom_validation.md

```diff
@@ -0,0 +1,102 @@
+# FormBuilder Custom Validation
+
+As useful as the built-in validators are, it is very useful to be able to include your own.
+Angular 2 allows you to do just that, with minimal effort.
+
+Let's assume we are using the same Login Form, but now we also want to test that our password has an exclamation mark somewhere in it.
+
+_app/login-form.component.ts_
+```ts
+function hasExclamationMark (input: FormControl) {
+  const hasExclamation = input.value.indexOf('!') >= 0;
+
+  return hasExclamation ? null : { needsExclamation: true };
```

Comment:
Personally, I've found that the mental agility needed to get to "a = b || c" might trip people up momentarily, but that `function maybeY () { if (x) return y; }` as a structural flow will confuse people at a much less localized level.

I'm totally happy to have that chat, but I've dealt with a lot of misery down the other path, which ends up influencing architecture decisions all the way across the app, sometimes.